content/en/continuous_integration/pipelines/teamcity.md

```diff
@@ -0,0 +1,104 @@
+    tag: "Documentation"
+    text: "Explore Pipeline Execution Results and Performance"
+  - link: "/continuous_integration/troubleshooting/"
+    tag: "Documentation"
+    text: "Troubleshooting CI"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+## Compatibility
+
+- **Supported TeamCity versions**: 2021.2 or later
+
+- **Partial retries**: View partially retried pipeline executions
+
+- **Queue time**: View amount of time pipeline jobs sit in the queue before processing
+
+- **Pipeline failure reasons**: Identify pipeline failure reasons from error messages
+
+## Configure the Datadog integration
+
+The integration between [TeamCity][1] and Datadog CI Visibility is provided through a TeamCity plugin.
+To activate the integration:
+1. Download the [Datadog CI Integration plugin][5] on the TeamCity server by going to
+**Administration** -> **Plugins** -> **Browse Plugin Repository**.
+2. If you don't already have one, add a [TeamCity composite build][6] as the last build of the build chain. This build must have
+a dependency on the current last build of the chain and no other builds depending on it. Build chains not having a
+composite build as their last build are ignored by the plugin. Below is an example of an expected build chain,
+in which `Aggregating Results` is the last composite build:
+{{< img src="ci/teamcity_build_chain.png" alt="TeamCity build chain with composite build at the end" style="width:90%;">}}
+The final composite build must be properly configured in terms of version control settings, with
+the VCS Root attached and the [VCS Trigger][13] configured.
+3. The following configuration parameters need to be present for TeamCity projects:
+   * `datadog.ci.api.key`: Your [Datadog API Key][2].
+   * `datadog.ci.site`: {{< region-param key="dd_site" code="true" >}}.
+   * `datadog.ci.enabled`: `true` (`false`
+   can be used to disable the plugin for a specific project).
+
+   These configuration parameters can be added either to TeamCity subprojects
+   or to the [TeamCity Root Project][10]. When added to the Root project, they are propagated
+   to all its subprojects. For example, to enable the plugin for all projects, add `datadog.ci.enabled` with the
+   value `true` to the Root Project. More information on defining configuration parameters
+   is available in the [TeamCity Project Hierarchy][9] documentation.
+4. To enable the plugin, click on **Enable uploaded plugins** in the **Administration** -> **Plugins** page.
+Alternatively, restart the TeamCity server.
+
+## Visualize pipeline data in Datadog
+
+View your data on the [Pipelines][3] and [Pipeline Executions][4] pages after the pipelines finish.
+
+**Note**: The Pipelines page only shows data for the [default branch][12] of each repository.
+
+## Troubleshooting
```

Comment:
Would it make sense to move the troubleshooting section to the end of the doc, just before the `Further reading` section? This is the format used in most other integration docs.

Reply:
Yeah makes sense! I've moved it